packages/compiler/test/server/vscode-settings.test.ts

```diff
@@ -0,0 +1,255 @@
+import { strictEqual } from "assert";
+import { beforeEach, describe, it, vi } from "vitest";
+import {
+  ClientConfigProvider,
+  createClientConfigProvider,
+} from "../../src/server/client-config-provider.js";
+
+describe("compiler: server: client-config-provider", () => {
+  let configProvider: ClientConfigProvider;
+
+  beforeEach(() => {
+    // Create a new client config provider before each test
+    configProvider = createClientConfigProvider();
+  });
+
+  describe("configuration API", () => {
+    it("should return undefined when configuration is missing without default", () => {
+      const result = configProvider.get("non.existent.key");
+      strictEqual(result, undefined);
+    });
+
+    it("should return default value when configuration is missing", () => {
+      const defaultValue = "default-value";
+      const result = configProvider.get("non.existent.key", defaultValue);
+      strictEqual(result, defaultValue);
+    });
+
+    it("should handle nested configuration keys", () => {
+      // Since configuration starts empty, test with default values
+      const result = configProvider.get("lsp.emit", ["default"]);
+      strictEqual(result[0], "default");
+    });
+
+    it("should handle boolean configuration values", () => {
+      const result = configProvider.get("formatting.enabled", false);
+      strictEqual(result, false);
+    });
+  });
+
+  describe("initialization and VS Code integration", () => {
+    let mockHost: any;
+
+    beforeEach(() => {
+      mockHost = {
+        log: vi.fn(),
+      } as any;
+    });
+
+    it("should handle initialization with mock connection and host", async () => {
+      // Create mock connection and host
+      const mockConnection = {
+        workspace: {
+          getConfiguration: vi.fn().mockResolvedValue({
+            lsp: { emit: ["openapi3"] },
+            formatting: { enabled: true },
+          }),
+        },
+        onDidChangeConfiguration: vi.fn(),
+      } as any;
+
+      await configProvider.initialize(mockConnection, mockHost);
+
+      // Verify configuration was fetched and applied
+      strictEqual(mockConnection.workspace.getConfiguration.mock.calls.length, 1);
+      strictEqual(mockConnection.workspace.getConfiguration.mock.calls[0]?.[0], "typespec");
+
+      // Verify configuration is now available - note that config is wrapped under 'typespec' key
+      const lspEmit = configProvider.get<string[]>("typespec.lsp.emit");
+      strictEqual(lspEmit?.[0], "openapi3");
+      strictEqual(configProvider.get("typespec.formatting.enabled"), true);
+
+      // Verify change handler was registered
+      strictEqual(mockConnection.onDidChangeConfiguration.mock.calls.length, 1);
+
+      // Verify logging - the detail should contain the wrapped settings
+      strictEqual(mockHost.log.mock.calls.length, 1);
```

Comment:
can you simplify those test this just feels a lot of uncessary things. You are only testing the config is resolved correctly the rest is irrelevant.